src/openms/include/OpenMS/KERNEL/FeatureMap.h

```diff
@@ -264,6 +264,19 @@ namespace OpenMS
     */
     OPENMS_DLLAPI void clear(bool clear_meta_data = true);
 
+    /// applies a function on all PeptideHits or only assigned ones
+    OPENMS_DLLAPI void applyFunctionOnPeptideHits(std::function<void(PeptideHit&)>& f, bool include_unassigned = true);
```

Comment:
std::function is SLOW...
probably better to use a function-template and pass in a lambda...